index.html

```diff
@@ -9222,6 +9222,240 @@ <h3 id="example-2">
+            new class, <code>EnveloppeFollower</code>, that instantiate a
+            special object that behaves like an <a>AudioNode</a>, described
+            below:
+          </p>
+          <pre>
+              var delay = new DelayNode(context, {delayTime: 0.006});
+              var gain = new GainNode(context);
+              var compression = new EnveloppeFollower();
+
+              input.connect(delay).connect(gain).connect(output);
+              input.connect(compression).connect(gain.gain);</pre>
+          <div class="note">
+            This implements the pre-delay and the application of the reduction
+            gain.
+          </div>
+          <p>
+            The following algorithm describes the
+            <code>EnveloppeFollower</code> object to be applied to the input
+            signal to produce the gain reduction value. First, a couple
+            variables that persist accross invocation of this algorithm, tied
+            to this particular <code>EnveloppeFollower</code> instance.
+          </p>
+          <ul>
+            <li>Let <var>detector average</var> be a floating point number,
+            initialized to 1.0.
+            </li>
+            <li>Let <var>compression gain</var> be a foating point number,
+            initialized to 1.0.
+            </li>
+          </ul>
+          <p>
+            Now, the actual algorithm to determine the reduction gain, in
+            linear gain value.
+          </p>
+          <ol>
+            <li>Let <var>input signal</var> be the the block of audio to be
+            processed.
+            </li>
+            <li>Let <var>threshold</var>, <var>knee</var> have the value of the
+            <a>AudioParam</a> of the same name, <a>converted to linear unit</a>
+            sampled at the time of processing of this block (as <a>k-rate</a>
+            parameters).
+            </li>
+            <li>Let <var>ratio</var> have the value of the <code>ratio</code>
+            <a>Audioparam</a>, sampled at the time of processing of this block
+            (as a <a>k-rate</a> parameter).
+            </li>
+            <li>Let <var>attack</var> and <var>release</var> have the value of
+            the <a>AudioParam</a> of the same name, sampled at the time of
+            processing (those are <a>k-rate</a> parameters), converted to
+            sample-frames, based on the sample-rate of the
+            <a>BaseAudioContext</a> this <a>DynamicsCompressorNode</a> is
+            <a href="#associated">associated</a> with.
+            </li>
+            <li>Set <var>attack</var> and <var>release</var> to the
+            multiplication between themselves and the sample-rate for this <a>
+              BaseAudioContext</a>.
+            </li>
+            <li>For each samples <var>input</var> of <var>input signal</var>,
+            execute the following steps:
+              <ol>
+                <li>Let <var>envelope rate</var> be the result of computing the
```

Comment:
`[[foo]]` refer to internal slots of objects (it's the ECMAScript notation). `<var></var>` is recommended for algorithm variables.